src/trace/dma-trace.c

```diff
@@ -210,6 +213,9 @@ static int dma_trace_buffer_init(struct dma_trace_data *d)
 
 	spin_unlock_irq(&d->lock, flags);
 
+	/* it should be the very first sent log for easily identification */
+	tr_info(&dt_tr, "FW ABI 0x%x DBG_ABI 0x%x hash 0x%x (ldc hash " META_QUOTE(SOF_FW_HASH) ")",
+		SOF_ABI_VERSION, SOF_ABI_DBG_VERSION, SOF_FW_HASH);
```

Comment:
This sort of trace is also very important to detect if the trace actually works on CI devices. I don't know if it's possible but we may even want to flush this first trace to guarantee it will come out, instead of waiting for additional traces to fill the DMA buffer, e.g. we should have e.g. a 1s timeout to flush all existing traces for debug.

Reply:
I didn't fully catch idea with timeout (flush + waiting for trace DMA to be done ?), but flush is a good idea

Reply:
the idea is that you flush the DMA buffer when
a) it reaches a watermark or
b) it contains data that has been in the buffer for more than N ms.

There are too many cases where we have incomplete trace information that we can only get by starting a new use case, just to flush the buffer.